pkg/yang/entry_test.go

```diff
@@ -1658,6 +1658,310 @@ func TestActionRPC(t *testing.T) {
 	}
 }
 
+var testIfFeatureModules = []struct {
+	name string
+	in   string
+}{
+	{
+		name: "if-feature.yang",
+		in: `module if-feature {
+  namespace "urn:if-feature";
+  prefix "feat";
+
+  container cont {
+    if-feature ft;
+    action act {
+      if-feature ft;
+    }
+  }
+
+  anydata data {
+    if-feature ft1;
+    if-feature ft2;
+  }
+
+  anyxml xml {
+    if-feature ft;
+  }
+
+  choice ch {
+    if-feature ft;
+    case cs {
+      if-feature ft;
+    }
+  }
+
+  feature f {
+    if-feature ft;
+  }
+
+  leaf l {
+    if-feature ft;
+    type bits {
+      bit A {
+        if-feature ft;
+      }
+    }
+  }
+
+  leaf-list ll {
+    if-feature ft;
+    type enumeration {
+      enum zero {
+        if-feature ft;
+      }
+    }
+  }
+
+  list ls {
+    if-feature ft;
+  }
+
+  notification n {
+    if-feature ft;
+  }
+
+  rpc r {
+    if-feature ft;
+  }
+
+  augment "/cont" {
+    if-feature ft;
+  }
+
+  identity id {
+    if-feature ft;
+  }
+
+  uses g {
+    if-feature ft-uses;
+    refine rf {
+      if-feature ft-refine;
+    }
+  }
+  grouping g {}
+}
+`,
+	},
+}
+
+func TestIfFeature(t *testing.T) {
+	entryIfFeatures := func(e *Entry) []*Value {
+		extra := e.Extra["if-feature"]
+		if len(extra) == 0 {
+			return nil
+		}
+		return extra[0].([]*Value)
+	}
+
+	ms := NewModules()
+	for _, tt := range testIfFeatureModules {
+		if err := ms.Parse(tt.in, tt.name); err != nil {
+			t.Fatalf("could not parse module %s: %v", tt.name, err)
+		}
+	}
+
+	if errs := ms.Process(); len(errs) > 0 {
+		t.Fatalf("could not process modules: %v", errs)
+	}
+
+	mod, _ := ms.GetModule("if-feature")
+
+	testcases := []struct {
+		name           string
+		inIfFeatures   []*Value
+		wantIfFeatures []string
+	}{
+		// Node statements
+		{
+			name:           "action",
+			inIfFeatures:   entryIfFeatures(mod.Dir["cont"].Dir["act"]),
+			wantIfFeatures: []string{"ft"},
```

Comment:
`ft-action` and etc below.

Yes, this really just makes it easier for reviewers to ensure that the test cases are matching specific test data without checking the code and attempting test breakage to ensure the appropriate labels are being referred to!  (Much appreciated)

Reply:
Done, along with all other similar comments.